File: LibraryProject_C++/Film.cpp
#include "Film.h"

Film::Film(const string& call_no,const string& title,const string& subjects,const string& director,const string& notes,const string& year):Media(call_no,title,subjects,notes),director(director),year(year)
{
}


Film::~Film(){}

/*
   compare_other finds if the input string exists in Notes, Director
   and year. It returns boolean value.
 */

bool Film::compare_other(const string& other_string) const{

	if(director.find(other_string)!=string::npos){
		return true;
	}
	else if(year.find(other_string)!=string::npos){
		return true;
	}
	else if(Media::compare_other(other_string)){
		return true;
	}
	else{
		return false;
	}
}

/*
   display function to print Film attributes. This function gives
   call to display() of Media class.
 */
void Film::display() const
{
	cout << "------Film------" << endl;

	Media::display();  
	cout<<"Director: "<<director<<endl;
	cout<<"Year:     "<<year<<endl<<endl;


}

File: libraryProject_Python/Film.py
from Media import Media

class Film(Media):

	def __init__(self,callNo,title,subjects,director,notes,year):
		self.callNo = callNo
		self.title = title
		self.subjects = subjects
		self.director = director
		self.notes = notes
		self.year = year
		super(Film,self).__init__(self.callNo,self.title,self.subjects,self.notes)


	
	def compare_Other(self,other):
		if other in self.director:
			return True
		elif other in self.year:
			return True
		elif other in self.notes:
			return True
		else:
			return False




	def display(self):
		super(Film,self).display()
		print ("Director: %s" % (self.director))
		print ("Year: %s" % (self.year))
		
		
